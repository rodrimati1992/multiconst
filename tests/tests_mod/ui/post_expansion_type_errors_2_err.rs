type Unit = ();

// ensure that mismatched tuple pattern and tuple type alias type error at macro expansion time
multiconst::multiconst!{
    const (A300, A301): Unit = ();
}

fn main(){}