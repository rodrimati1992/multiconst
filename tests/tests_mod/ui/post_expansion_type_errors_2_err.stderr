error[E0277]: the trait bound `(): FieldType<Usize<0_usize>>` is not satisfied
 --> tests/tests_mod/ui/post_expansion_type_errors_2_err.rs:5:12
  |
5 |     const (A300, A301): Unit = ();
  |            ^^^^ the trait `FieldType<Usize<0_usize>>` is not implemented for `()`
  |
  = note: required because of the requirements on the impl of `FieldType<(Usize<0_usize>,)>` for `()`
  = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `(): FieldType<Usize<1_usize>>` is not satisfied
 --> tests/tests_mod/ui/post_expansion_type_errors_2_err.rs:5:18
  |
5 |     const (A300, A301): Unit = ();
  |                  ^^^^ the trait `FieldType<Usize<1_usize>>` is not implemented for `()`
  |
  = note: required because of the requirements on the impl of `FieldType<(Usize<1_usize>,)>` for `()`
  = note: this error originates in the macro `multiconst::multiconst` (in Nightly builds, run with -Z macro-backtrace for more info)
